shell.nix

```diff
@@ -18,7 +18,12 @@ pkgs.mkShell {
   buildInputs = [
     (daml { stdenv = pkgs.stdenv;
             jdk = pkgs.openjdk11_headless;
-            version = damlYaml.sdk-version; })
+            curl = pkgs.curl;
+            curl_cert = "${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt";
+            version = damlYaml.sdk-version;
+            os = if pkgs.stdenv.isDarwin then "macos" else "linux";
+            hashes = { linux = "sha256-1P2SAwgVI3IXKaSSKTAbao8wN/eQaQKNuPcCYaGAzAQ=";
+                       macos = "sha256-kXUWD2/0vLnv0I9lSZEfqFh3JjEvIeVBbafcC6OEteQ="; };})
```

Comment:
I presume these hashes must equal the hash of the installed `daml` installation by nix?

Reply:
Yes; nix tells you the values it gets.

Reply:
Perhaps I don't understand how this works - are these the hashes Nix validates against the install? Ie, we'd have to know the hashes in advance in order for your changes to work?

Reply:
I attempted to implement the changes in [this pull request](https://github.com/digital-asset/daml-finance/pull/1165) but encountered issues. I realize that I'm unable to build this branch locally as well, I see an error message that `ARTIFACTORY_USERNAME and ARTIFACTORY_PASSWORD must be set` although I have set both in my `.envrc.private` file. I also tried to modify the script to accept an `ARTIFACTORY_TOKEN`. @brianweir-da, would be great if we could look into it briefly tomorrow?

Reply:
I've moved my work over to the 2.8 branch in https://github.com/digital-asset/daml-finance/pull/1166. The `update-daml-hashes` file over there will give you more details, but the gist of it is Nix needs to know what the output of a derivation is in order to grant that derivation network access. So we have to give it the hash of the output, i.e. in this case the sha256sum of the tarball we're downloading.

Reply:
@garyverhaegen-da seems I lack access to the latest artifactory snapshot, I have sent an email to the help desk.